Exception creating count queries for Spring repositories

### Description
Using `BlazePersistenceRepositoryFactory` with Spring Boot 2.0.4.RELEASE and Hibernate 5.2 integration, a repository with a count query throws an exception at app startup.

### Expected behavior
No exception is thrown and the app starts with a working repository.

### Actual behavior
Using `BlazePersistenceRepositoryFactory` with Spring Boot 2.0.4.RELEASE and Hibernate 5.2 integration, a repository with a count query such as

```java
interface MyRepository extends Repository<MyEntity, String> {

   long countByAttribute(String attribute);

}
```

throws the following exception on app startup:

```
java.lang.IllegalArgumentException: Could not create query metamodel for method public abstract long UserRepository.countByPrimaryEmailAddress(java.lang.String)!
    at com.blazebit.persistence.spring.data.impl.repository.BlazePersistenceRepositoryFactory$CreateQueryLookupStrategy.resolveQuery(BlazePersistenceRepositoryFactory.java:190)
    at com.blazebit.persistence.spring.data.impl.repository.BlazePersistenceRepositoryFactory$CreateIfNotFoundQueryLookupStrategy.resolveQuery(BlazePersistenceRepositoryFactory.java:229)
    at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.lookupQuery(RepositoryFactorySupport.java:553)
    at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.lambda$mapMethodsToQuery$1(RepositoryFactorySupport.java:546)
Caused by: java.lang.IllegalArgumentException: DynamicSort must not be null!
    at org.springframework.util.Assert.notNull(Assert.java:193)
    at org.springframework.data.repository.query.parser.AbstractQueryCreator.createQuery(AbstractQueryCreator.java:94)
    at com.blazebit.persistence.spring.data.base.query.AbstractPartTreeBlazePersistenceQuery$CountQueryPreparer.invokeQueryCreator(AbstractPartTreeBlazePersistenceQuery.java:435)
    at com.blazebit.persistence.spring.data.base.query.AbstractPartTreeBlazePersistenceQuery$QueryPreparer.<init>(AbstractPartTreeBlazePersistenceQuery.java:224)
    at com.blazebit.persistence.spring.data.base.query.AbstractPartTreeBlazePersistenceQuery$CountQueryPreparer.<init>(AbstractPartTreeBlazePersistenceQuery.java:400)
    at com.blazebit.persistence.spring.data.base.query.AbstractPartTreeBlazePersistenceQuery.<init>(AbstractPartTreeBlazePersistenceQuery.java:82)
    at com.blazebit.persistence.spring.data.impl.query.PartTreeBlazePersistenceQuery.<init>(PartTreeBlazePersistenceQuery.java:38)
    at com.blazebit.persistence.spring.data.impl.repository.BlazePersistenceRepositoryFactory$CreateQueryLookupStrategy.resolveQuery(BlazePersistenceRepositoryFactory.java:187)
    ... 154 more
```

### Steps to reproduce

1. Create a new Spring project and add Blaze deps.
2. Configure the app to use `BlazePersistenceRepositoryFactory` for repositories.
3. Create a repository interface with a count query.
4. Start the app.

Example project: https://github.com/heruan/spring-blaze

### Environment
Version: 1.3.0-Alpha2
JPA-Provider: Hibernate 5.2
Spring Boot: 2.0.4.RELEASE

Worth noting that with Spring Boot 2.1.0.M1 and Hibernate 5.3 integration, the exception is not thrown.

--- a/integration/spring-data/base/src/main/java/com/blazebit/persistence/spring/data/base/query/AbstractPartTreeBlazePersistenceQuery.java
+++ b/integration/spring-data/base/src/main/java/com/blazebit/persistence/spring/data/base/query/AbstractPartTreeBlazePersistenceQuery.java
@@ -93,7 +93,6 @@ public abstract class AbstractPartTreeBlazePersistenceQuery extends AbstractJpaQ
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public TypedQuery<Long> doCreateCountQuery(Object[] values) {
         throw new UnsupportedOperationException();
     }
@@ -432,7 +431,7 @@ public abstract class AbstractPartTreeBlazePersistenceQuery extends AbstractJpaQ
 
         @Override
         protected CriteriaQuery<?> invokeQueryCreator(FixedJpaQueryCreator creator, Sort sort) {
-            return creator.createQuery(sort);
+            return creator.createQuery();
         }
     }
 }

--- a/integration/spring-data/base/src/main/java/com/blazebit/persistence/spring/data/base/query/JpaParameters.java
+++ b/integration/spring-data/base/src/main/java/com/blazebit/persistence/spring/data/base/query/JpaParameters.java
@@ -53,7 +53,7 @@ public class JpaParameters extends Parameters<JpaParameters, JpaParameter> {
         super(parameters);
     }
 
-	/* *
+    /**
      * Gets the parameters annotated with {@link OptionalParam}.
      *
      * @return the optional parameters
@@ -72,13 +72,13 @@ public class JpaParameters extends Parameters<JpaParameters, JpaParameter> {
     }
 
     /*
-	 * (non-Javadoc)
-	 * @see org.springframework.data.repository.query.Parameters#createParameter(org.springframework.core.MethodParameter)
-	 */
-	@Override
-	protected JpaParameter createParameter(MethodParameter parameter) {
-		return new JpaParameter(parameter);
-	}
+     * (non-Javadoc)
+     * @see org.springframework.data.repository.query.Parameters#createParameter(org.springframework.core.MethodParameter)
+     */
+    @Override
+    protected JpaParameter createParameter(MethodParameter parameter) {
+        return new JpaParameter(parameter);
+    }
 
     /*
      * (non-Javadoc)
@@ -111,22 +111,24 @@ public class JpaParameters extends Parameters<JpaParameters, JpaParameter> {
 
             super(parameter);
 
-			this.parameter = parameter;
-			this.optional = parameter.getParameterAnnotation(OptionalParam.class);this.annotation = parameter.getParameterAnnotation(Temporal.class);
-			this.temporalType = null;
+            this.parameter = parameter;
+            this.optional = parameter.getParameterAnnotation(OptionalParam.class);
+            this.annotation = parameter.getParameterAnnotation(Temporal.class);
+            this.temporalType = null;
 
-			if (!isDateParameter() && hasTemporalParamAnnotation()) {
-				throw new IllegalArgumentException(
-						Temporal.class.getSimpleName() + " annotation is only allowed on Date parameter!");
-			}
-		}
+            if (!isDateParameter() && hasTemporalParamAnnotation()) {
+                throw new IllegalArgumentException(
+                    Temporal.class.getSimpleName() + " annotation is only allowed on Date parameter!");
+            }
+        }
 
-		public String getParameterName() {
-			Param annotation = parameter.getParameterAnnotation(Param.class);
-			if (annotation != null) {return annotation.value();
+        public String getParameterName() {
+            Param annotation = parameter.getParameterAnnotation(Param.class);
+            if (annotation != null) {
+                return annotation.value();
             }
             return optional == null ? parameter.getParameterName() : optional.value();
-		}
+        }
 
         /*
          * (non-Javadoc)
@@ -137,19 +139,21 @@ public class JpaParameters extends Parameters<JpaParameters, JpaParameter> {
             return super.isBindable() || isTemporalParameter();
         }
 
-		@Override
+        @Override
         public boolean isSpecialParameter() {
             return super.isSpecialParameter() || isOptionalParameter();
         }
 
         boolean isOptionalParameter() {
             return optional != null;
-        }/**
-		 * @return {@literal true} if this parameter is of type {@link Date} and has an {@link Temporal} annotation.
-		 */
-		boolean isTemporalParameter() {
-			return isDateParameter() && hasTemporalParamAnnotation();
-		}
+        }
+
+        /**
+         * @return {@literal true} if this parameter is of type {@link Date} and has an {@link Temporal} annotation.
+         */
+        boolean isTemporalParameter() {
+            return isDateParameter() && hasTemporalParamAnnotation();
+        }
 
         /**
          * @return the {@link TemporalType} on the {@link Temporal} annotation of the given {@link Parameter}.

